docs/client_libraries.md

```diff
@@ -0,0 +1,286 @@
+without Docker you must first install those dependencies. This section
+describes the client build for Ubuntu 18.04 and Windows 10
+systems.
+
+To build the libraries using CMake, first change directory to the root
+of the repo and checkout the release version of the branch that you
+want to build (or the master branch if you want to build the
+under-development version).
+
+```bash
+$ git checkout r20.09
+```
+
+#### Ubuntu 18.04
+
+For Ubuntu, the dependencies and how to install them can be found in
+Dockerfile.client. Also note that the dependency name may be different
+depending on the version of the system.
+
+To build on Ubuntu, run the following to configure and build:
+
+```bash
+$ mkdir builddir && cd builddir
+$ cmake -DCMAKE_BUILD_TYPE=Release ../build
+$ make -j8 client
+```
+
+If you want to build a version of the client libraries and examples
+that does not include the CUDA shared memory support, use the
+following cmake configuration.
+
+```bash
+$ cmake -DTRITON_ENABLE_GPU=OFF -DTRITON_ENABLE_METRICS_GPU=OFF -DCMAKE_BUILD_TYPE=Release ../build
+```
+
+When the build completes the libraries can be found in
+client/install/lib, the corresponding headers in
+client/install/include, and the Python wheel files in
+client/install/python. The client/install directory will also contain
+the built client examples that you can learn more about in
+[Examples](client_examples.md).
+
+#### Windows 10
+
+For Windows, the dependencies can be installed using pip and
+[vcpkg](https://github.com/Microsoft/vcpkg) which is a C++ library
+management tool on Windows. The following shows how to install the
+dependencies using them, and you can also install the dependencies in
+other ways that you prefer.
+
+```
+> .\vcpkg.exe install openssl:x64-windows zlib:x64-windows rapidjson:x64-windows
+> .\pip.exe install --upgrade setuptools grpcio-tools wheel
+```
+
+The vcpkg step above installs openssl, zlib and rapidjson,
+":x64-windows" specifies the target and it is optional. The path to
+the libraries should be added to environment variable "PATH", by
+default it is \path\to\vcpkg\installed\\\<target>\bin. Update the
+pip to get the proper wheel from PyPi. Users may need to invoke
+pip.exe from a command line ran as an administrator.
+
+To build the client for Windows, as there is no default
+build system available, you will need to specify the generator for
+CMake to match the build system you are using. For instance, if you
+are using Microsoft Visual Studio, you should do the following.
+
+```
+> cd build
+> cmake -G"Visual Studio 16 2019" -DCMAKE_BUILD_TYPE=Release
+> MSBuild.exe client.vcxproj -p:Configuration=Release
+```
+
+If you want to build a version of the client libraries and examples
+that does not include the CUDA shared memory support, use the
+following cmake configuration.
+
+```
+> cmake -G"Visual Studio 16 2019" -DTRITON_ENABLE_GPU=OFF -DTRITON_ENABLE_METRICS_GPU=OFF -DCMAKE_BUILD_TYPE=Release
```

Comment:
Maybe we can include building Windows client with a non-default branch for `common` repo?